File: days/10/part2.js
const data = require('fs')
    .readFileSync('input', 'utf8')
    .split(/\n/)

var res = data.map((line)=>{
    var stack = []
    for( c in line){
        switch (line[c]) {
            case '{': 
            case '(': 
            case '[': 
            case '<': 
                stack.push(line[c])
                break;
            case '}':
                if (stack.at(-1) == '{') stack.pop()
                else return 1197
                break;
            case ')':
                if (stack.at(-1) == '(') stack.pop()
                else return 3
                break;
            case ']':
                if (stack.at(-1) == '[') stack.pop()
                else return 57
                break;
            case '>':
                if (stack.at(-1) == '<') stack.pop()
                else return 25137
                break;
        }
    }
    return stack
})
.filter(x=>!Number.isInteger(x))
.map((line)=>{
    return line.reduceRight((sum,c,i)=>{
        switch(c) {
            case '(': return (sum * 5) + 1
            case '[': return (sum * 5) + 2
            case '{': return (sum * 5) + 3
            case '<': return (sum * 5) + 4
        }
    },0)
}, 0)
res.sort((a,b)=> b-a)
console.log(res[(res.length-1)/2])
// .reduce((a,b)=>a+b))

File: 2021/6/part2.rb
school = Array.new(9,0)
input = IO.readlines("input")
input = input[0].split(',')

for i in 0...input.size do
    school[Integer(input[i])] += 1
end

257.times do 
    school[9] = school[0]
    for i in 0..8 do
        school[i] = school[i+1]
    end
    school[6] += school[9]

end

puts school[0,8].reduce(0) {|sum, e| sum + e}


